pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -2083,12 +2083,19 @@ func (dm *DockerManager) computePodContainerChanges(pod *v1.Pod, podStatus *kube
 
 	// check the status of the containers
 	for index, container := range pod.Spec.Containers {
-
 		containerStatus := podStatus.FindContainerStatusByName(container.Name)
-		if containerStatus == nil || containerStatus.State != kubecontainer.ContainerStateRunning {
+		if containerStatus == nil {
+			if kubecontainer.ShouldContainerBeRestarted(&container, pod, podStatus) {
+				// If we are here it means that the container never existed and should be created or was GC'd.
+				message := fmt.Sprintf("Container %+v has not been created, and will create it soon.", container)
+				glog.V(3).Info(message)
+				containersToStart[index] = message
```

Comment:
Let's not duplicate this code. I suggest keeping the original conditional, and wrapping the message statement in a new conditional.